infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsLogEntryReader.java

```diff
@@ -98,11 +103,22 @@ public StreamsLogEntryReader(CorfuRuntime runtime, LogReplicationConfig config)
         txOpaqueStream = new TxOpaqueStream(rt);
     }
 
-    private LogReplicationEntry generateMessageWithOpaqueEntryList(List<OpaqueEntry> opaqueEntryList, UUID logEntryRequestId) {
+    private LogReplicationEntryMsg generateMessageWithOpaqueEntryList(
+            List<OpaqueEntry> opaqueEntryList, UUID logEntryRequestId) {
         // Set the last timestamp as the max timestamp
         long currentMsgTs = opaqueEntryList.get(opaqueEntryList.size() - 1).getVersion();
-        LogReplicationEntry txMessage = new LogReplicationEntry(MSG_TYPE, topologyConfigId, logEntryRequestId,
-                currentMsgTs, preMsgTs, globalBaseSnapshot, sequence, opaqueEntryList);
+        LogReplicationEntryMetadataMsg metadata = LogReplicationEntryMetadataMsg.newBuilder()
+                .setEntryType(MSG_TYPE)
+                .setTopologyConfigID(topologyConfigId)
+                .setSyncRequestId(getUuidMsg(logEntryRequestId))
+                .setTimestamp(currentMsgTs)
+                .setPreviousTimestamp(preMsgTs)
+                .setSnapshotTimestamp(globalBaseSnapshot)
+                .setSnapshotSyncSeqNum(sequence)
+                .build();
+
+        LogReplicationEntryMsg txMessage = getLrEntryMsg(ByteString.copyFrom(generatePayload(opaqueEntryList)), metadata);
```

Comment:
Here as well.

Reply:
Done as described in https://github.com/CorfuDB/CorfuDB/pull/2858#discussion_r573372305